assisted-inject-processor/src/main/java/com/squareup/inject/assisted/processor/AssistedInjectProcessor.kt

```diff
@@ -244,7 +252,7 @@ class AssistedInjectProcessor : AbstractProcessor() {
       throw StopProcessingException("Factory interface defines multiple factory methods.", factory)
     }
     val method = methods.single()
-    if (!types.isAssignable(type.asType(), method.returnType)) {
+    if (!types.isAssignable(types.erasure(type.asType()), types.erasure(method.returnType))) {
```

Comment:
Yes, it will fail to compile. It'd still be nice to surface the error correctly. If javac can figure out that types aren't assignable we should be able to somehow.

If you want to comment out the check and `@Ignore` any test for now that's fine.